src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/CloseTextTagOnAutoInsertProvider.cs

```diff
@@ -68,11 +75,15 @@ public override bool TryResolveInsertion(Position position, FormattingContext co
             return true;
         }
 
-        private static bool IsAtTextTag(FormattingContext context, Position position)
+        private static bool IsAtTextTag(FormattingContext context, Position position, ILogger logger)
         {
             var syntaxTree = context.CodeDocument.GetSyntaxTree();
 
-            var absoluteIndex = position.GetAbsoluteIndex(context.SourceText) - 1;
+            if (!position.TryGetAbsoluteIndex(context.SourceText, out var absoluteIndex, logger))
+            {
+                return false;
+            }
+            absoluteIndex -= 1;
```

Comment:
nit: In the Roslyn repo we have an analyzer code style rule where a new line is required after an if-statement. I'm not sure if we want to adopt the same thing in Razor but I feel like it could be good for consistency

Reply:
I'm 100% for it. Also if there's one to avoid double-newlines. Basically I want my build to fail if anyone is going to leave me comments about my messy formatting lol.

Reply:
Yes, both of those are built in to VS, we just need to turn the severity up (though strictly, its newline after close brace)

Reply:
Do you know the code? I went looking but didn't find it.

Reply:
I'm using this in Trains.NET:
```
# dotnet_style_allow_multiple_blank_lines_experimental
dotnet_diagnostic.IDE2000.severity = warning

# csharp_style_allow_embedded_statements_on_same_line_experimental
dotnet_diagnostic.IDE2001.severity = none

# csharp_style_allow_blank_lines_between_consecutive_braces_experimental
dotnet_diagnostic.IDE2002.severity = warning

# dotnet_style_allow_statement_immediately_after_block_experimental
dotnet_diagnostic.IDE2003.severity = suggestion

# csharp_style_allow_blank_line_after_colon_in_constructor_initializer_experimental
dotnet_diagnostic.IDE2004.severity = warning
```